addons/stock/report/product_templates.xml

```diff
@@ -4,18 +4,18 @@
         <template id="label_product_product_view">
 		<t t-foreach="quantity.items()" t-as="barcode_and_qty_by_product">
                 <t t-set="product" t-value="barcode_and_qty_by_product[0]"/>
-                <t t-foreach="barcode_and_qty_by_product[1]" t-as="barcode_and_qty">
-                    <t t-set="barcode" t-value="barcode_and_qty[0]"/>
-                    <t t-foreach="range(barcode_and_qty[1])" t-as="qty">
+                <t t-foreach="barcode_and_qty_by_product[1]" t-as="product_info">
+                    <t t-set="barcode" t-value="product_info['barcode']"/>
+                    <t t-foreach="range(product_info['quantity'])" t-as="qty">
                         <t t-translation="off">
-^XA
-^FT100,80^A0N,40,30^FD<t t-esc="product.display_name"/>^FS
+^XA^CI28

```

Comment:
Since we're adding `^CI28` here, I think we should remove the one added in line 22 as it's already included now.
As per their [documentation](https://labelary.com/faq.html#i18n):
> Be sure to remove any other ^CI commands in your ZPL template.